airbyte-integrations/connectors/source-e2e-test/src/main/java/io/airbyte/integrations/source/e2e_test/InfiniteFeedSource.java

```diff
@@ -63,11 +64,13 @@ public AirbyteCatalog discover(final JsonNode config) {
       protected AirbyteMessage computeNext() {
         if (anotherRecordPredicate.test(i.get())) {
           if (i.get() != 0) {
-            try {
-              LOGGER.info("sleeping for {} ms", sleepTime);
-              sleep(sleepTime);
-            } catch (final InterruptedException e) {
-              throw new RuntimeException();
+            if (sleepTime.isPresent()) {
+              try {
+                LOGGER.info("sleeping for {} ms", sleepTime.get());
+                sleep(sleepTime.get());
+              } catch (final InterruptedException e) {
+                throw new RuntimeException();
```

Comment:
Nit: we can provide more information in the stack trace using `throw new RuntimeException(e);`

Reply:
Ah good catch